activation/nipost_test.go

```diff
@@ -807,6 +807,344 @@ func TestNIPSTBuilder_PoetUnstable(t *testing.T) {
 	})
 }
 
+// TestNIPoSTBuilder_PoETConfigChange checks if
+// it properly detects added/deleted PoET services and re-registers if needed.
+func TestNIPoSTBuilder_PoETConfigChange(t *testing.T) {
+	t.Parallel()
+
+	sig, err := signing.NewEdSigner()
+	require.NoError(t, err)
+
+	challenge := types.NIPostChallenge{
+		PublishEpoch: postGenesisEpoch + 2,
+	}
+
+	const (
+		poetProverAddr  = "http://localhost:9999"
+		poetProverAddr2 = "http://localhost:9988"
+	)
+
+	t.Run("1 poet deleted BEFORE round started -> continue get proof from known poets", func(t *testing.T) {
+		db := localsql.InMemory()
+
+		require.NoError(t, nipost.AddChallenge(db, sig.NodeID(), &challenge))
+		challengeHash := wire.NIPostChallengeToWireV1(&challenge).Hash()
+
+		ctrl := gomock.NewController(t)
+		poetProver := defaultPoetServiceMock(t, ctrl, poetProverAddr)
+		poetProver.EXPECT().Proof(gomock.Any(), "1").AnyTimes().Return(
```

Comment:
Using `AnyTimes()` is rarely useful. We should be certain which calls are expected. Please avoid it if possible
```suggestion
		poetProver.EXPECT().Proof(gomock.Any(), "1").Return(
```